### DISCUSSION TITLE
Aggregation By Field

### DISCUSSION THREAD

**Discussion by BishoyNarouz** [prisma/prisma#6427] [Q&A] [answered]
```prisma
model Application {
id
name
status      completed || rejected || canceled
description
}
```
how can i aggregate results by status field?
example: 
```js
{
completed: [
id 
name
description
],
canceled : [
id 
name
description
]
........
}

prisma.application.findMany({})
```

**Comment by BishoyNarouz** [comment]
i found the solution in this link:
https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by
but i could not return the result like that:
```js
{
completed: [
id
name
description
],
canceled : [
id
name
description
]
........
}
```

**Reply by BishoyNarouz** [reply]
@ryands17

**Comment by pantharshit00** [comment, marked as answer]
Hey @BishoyNarouz 

You can do something like this:
```js
const statuses = await prisma.application.groupBy({
    by: ["status"],
  });

  const statusByCategoryPromises = statuses.map((s) =>
    prisma.application.findMany({ where: { status: s.status } })
  );

  const statusByCategory = await Promise.all(statusByCategoryPromises);

  const data = statusByCategory.map((s, i) => ({ [statuses[i].status]: s }));

  console.dir(data, { depth: Infinity });
  ```
  
  This above will give you the desired result.

**Reply by BishoyNarouz** [reply]
thanks @pantharshit00

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
